assets/dev/js/frontend/handlers/container/grid-outline.js

```diff
@@ -0,0 +1,229 @@
+export default class GridOutline extends elementorModules.frontend.handlers.Base {
+	isActive() {
+		return elementorFrontend.isEditMode();
+	}
+
+	getDefaultSettings() {
+		return {
+			selectors: {
+				gridOutline: '.e-grid-outline',
+				directGridOverlay: ':scope > .e-grid-outline',
+				boxedContainer: ':scope > .e-con-inner',
+			},
+			classes: {
+				outline: 'e-grid-outline',
+				outlineItem: 'e-grid-outline-item',
+			},
+		};
+	}
+
+	getDefaultElements() {
+		const selectors = this.getSettings( 'selectors' );
+
+		return {
+			outlineParentContainer: null,
+			gridOutline: this.findElement( selectors.gridOutline ),
+			directChildGridOverlay: this.findElement( selectors.directGridOverlay ),
+		};
+	}
+
+	onInit() {
+		super.onInit();
+
+		this.initLayoutOverlay();
+	}
+
+	bindEvents() {
+		elementorFrontend.elements.$window.on( 'resize', this.onDeviceModeChange.bind( this ) );
+		this.addChildLifeCycleEventListeners();
+	}
+
+	unbindEvents() {
+		this.removeChildLifeCycleEventListeners();
+		elementorFrontend.elements.$window.off( 'resize', this.onDeviceModeChange.bind( this ) );
+	}
+
+	initLayoutOverlay() {
+		this.getContainer();
+
+		if ( ! this.elements.outlineParentContainer ) {
+			return;
+		}
+
+		this.removeExistingOverlay();
+		this.createOverlayContainer();
+		this.createOverlayItems();
+	}
+
+	getContainer() {
+		const { grid_outline: gridOutline } = this.getElementSettings();
+
+		this.elements.outlineParentContainer = null;
```

Comment:
@davseve is this line needed? It is already being set to `null` in `getDefaultElements()`?

Reply:
AFAIU, the value is set to null only in the first run, 
After that each run will return the value we that was set before.
That is why I re set `this.elements.outlineParentContainer` to `null`